### PR TITLE
Fix/revise es6 import lessons

### LINKED ISSUE #17436: ES6 - Understand the Differences Between import and require - correct syntax not passing
#### test not passing
in this challenge, the final "NOTE" provided suggests using "./" if the file containing functions to import it are in the same folder, however the solution using that hint does not pass. 

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/understand-the-differences-between-import-and-require/

#### Tell us about your browser and operating system

* Browser Name: Chrome
* Browser Version: Version 66.0.3359.181 (Official Build) (64-bit)
* Operating System: High Sierra 10.13.4

#### If possible, add a screenshot here (not a screenshot, but this code does not pass) and it should, and is 'more correct' than "string_functions" since many compilers will give an error without the "./"
```
"use strict";
import { capitalizeString } from "./string_functions"
capitalizeString("hello!");

```

### PR DESCRIPTION
**Pull request by moT01**
This revises the es6 import lessons. There was a lot of invalid syntax in the current lessons, and some inaccurate information.

- I created the "Create a module script" challenge at the beginning
- I renamed  "Use export to Reuse a Code Block" to "Use export to Share a Code Block" and moved it to be taught before import - pretty much also rewrote this one
- I renamed "understand the differences between import and require" to "Reuse Javascript Code Using import" and moved it to be taught after the export - pretty much also rewrote this one
- "create an export fallback with export default" and "import a default export" and "Use * to Import Everything from a File" I tried not to modify these two too much - slight rewording - convert to backticks - fix up tests.

Hopefully I didn't make too many unnecessary modifications - but I think this is better more accurate.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Related to #36063
Closes https://github.com/freeCodeCamp/freeCodeCamp/issues/17436

### CODE REVIEW

**Review by thecodingaviator — COMMENTED** [review]
~@moT01 If you're done, please squash commits. Makes it easier as you already know~

File: curriculum/challenges/_meta/es6/meta.json
Comment (ojeytonwilliams) [inline curriculum/challenges/_meta/es6/meta.json]:
```suggestion
```

File: curriculum/challenges/_meta/es6/meta.json
Comment (ojeytonwilliams) [inline curriculum/challenges/_meta/es6/meta.json]:
```suggestion
```

File: curriculum/challenges/_meta/es6/meta.json
Comment (ojeytonwilliams) [inline curriculum/challenges/_meta/es6/meta.json]:
```suggestion
```

File: curriculum/challenges/_meta/es6/meta.json
Comment (ojeytonwilliams) [inline curriculum/challenges/_meta/es6/meta.json]:
```suggestion
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Nice PR!  Everything looks good, except for the meta duplication.  Literally the only thing I thought could have been done differently was to allow `export { subtract as default };` as a solution.  However, I appreciate we can't cover everything, so perhaps it's better as is.

Regardless, if you sort out meta.json, I'm happy to approve this.

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
LGTM :+1:

**Comment by RandellDawson** [comment]
@moT01 Overall I think you have made significant improvements to the existing challenges and added some valid new challenges.

Regarding the use of exporting functions using `default` and importing them, I think one thing missing is that a function does not even need a name in the `export default`.  It can be an anonymous function.  Also, the import can be any name when importing a default function.  The way the challenges are currently written, it implies there must be a function name specified in the `export default` and that when importing a default function, that it must be referenced by the same name.  

I am not exactly sure how best to add this verbiage to make it clear, but I believe the `export default` challenge should allow the user to not specify a named function and the challenge dealing with importing a default function should make it clear the function could be named something different if the export had a named function.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-an-export-fallback-with-export-default.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-an-export-fallback-with-export-default.english.md]:
```suggestion
Here are examples using <code>export default</code>:
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
New changes look good.  The `Here is a couple examples` seemed like incorrect grammar.  We can avoid that with my suggestion.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
@scissorsneedfoodtoo Just waiting on @ojeytonwilliams final review of this PR, but we also need a core team member to approve before we merge.

**Review by ojeytonwilliams — APPROVED** [review]
Sorry for the delay, LGTM!

**Review by scissorsneedfoodtoo — APPROVED** [review]
Everything LGTM - thank you @moT01 :+1: :+1:

**Pull request merged**
